### PR TITLE
Add config functionality for basic operations

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR adds `cfg->copyf()` calls for some basic functions like insert, find and so on in the linked list implementation.

This ensures that the user can do refcounting and other fancy things automatically on insertion/... of elements.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Targets #277 .

File: src/libreset/ll/base.c [hunk 1]
```diff
@@ -99,7 +104,7 @@ ll_find(
 
     while (iter) {
         if (cfg->cmpf(iter->data, d)) {
-            return iter->data;
+            return (!!cfg->copyf) ? cfg->copyf(iter->data) : iter->data;
```
Comment (neithernut) [inline src/libreset/ll/base.c]: Find must not copy.

**Comment by matthiasbeyer** [comment]
Question: Should we add more callbacks in the config for this? Something like
- `pre_insert()`
- `post_find()`
- `pre_delete()`

maybe even more?

**Comment by neithernut** [comment]
Nah.

**Comment by matthiasbeyer** [comment]
Nah?

**Comment by neithernut** [comment]
Would be bloat.

**Comment by matthiasbeyer** [comment]
Would be _a_ bloat, right?

**Comment by neithernut** [comment]
Boats?

**Comment by matthiasbeyer** [comment]
Goats!

**Comment by matthiasbeyer** [comment]
Floating Goats, actually.

**Comment by neithernut** [comment]
Meh.

**Comment by matthiasbeyer** [comment]
You got it.

**Comment by matthiasbeyer** [comment]
So, will this get merged now or what?

**Comment by neithernut** [comment]
Mergeldy merge.

**Pull request merged**